### PR TITLE
[Do not merge] Appenderator stuff

### PR DESCRIPTION
**Pull request by gianm**
stuff related to #1642 -- PR for discussion, not yet suitable for merging. based off of #1896, as this PR depends on that one.

### CODE REVIEW

File: common/src/main/java/io/druid/metadata/MetadataStorageTablesConfig.java [hunk 1]
```diff
@@ -67,6 +73,8 @@ public static MetadataStorageTablesConfig fromBase(String base)
   @JsonCreator
   public MetadataStorageTablesConfig(
       @JsonProperty("base") String base,
+      @JsonProperty("dataSource") String dataSourceTable,
```
Comment (drcrallen) [inline common/src/main/java/io/druid/metadata/MetadataStorageTablesConfig.java]: This list is getting kind of silly... we may want to consider a better way of handling this.

Comment (drcrallen) [inline common/src/main/java/io/druid/metadata/MetadataStorageTablesConfig.java]: ((not intended to be a blocker here))

**Comment by himanshug** [comment]
thanks, this clarifies some of the doubts I had while reviewing #1896

**Comment by gianm** [comment]
superseded by #2220

**Pull request closed without merging**